lib/uvobot/notifications/discourse_notifier.rb

```diff
@@ -32,23 +32,29 @@ def matching_announcements_found(_page_info, announcements)
 
       def announcement_to_topic(announcement)
         detail = @scraper.get_announcement_detail(announcement[:link][:href])
+        body_messages = ["**Obstarávateľ:** #{announcement[:procurer]}",
+                         "**Predmet obstarávania:** #{announcement[:procurement_subject]}",
+                         detail_messages(detail),
+                         "**Zdroj:** [#{announcement[:link][:text]}](#{announcement[:link][:href]})"]
 
         {
           title: announcement[:procurement_subject].to_s,
-          body: ["**Obstarávateľ:** #{announcement[:procurer]}",
-                 "**Predmet obstarávania:** #{announcement[:procurement_subject]}",
-                 detail_message(detail),
-                 "**Zdroj:** [#{announcement[:link][:text]}](#{announcement[:link][:href]})"].join("  \n")
+          body: body_messages.flatten(1).join("  \n")
         }
       end
 
-      def detail_message(detail)
+      def detail_messages(detail)
         if detail
-          "**Cena:** #{detail[:amount]}"
+          ["**Cena:** #{fallback_if_nil(detail[:amount])}",
```

Comment:
Potom tam ta hlaska o nepodarenej extrakcii bude pre kazdy kluc. Do metody som to dal aby to bolo na jednom mieste.